### DISCUSSION TITLE
Parsing cue constraints

### DISCUSSION THREAD

**Discussion by srikrsna** [cue-lang/cue#1094] [Q&A]
Hey everyone, I am trying to use Cue for validation. I wanted to built a protoc plugin that would generate validation functions based on `cue.value`  field option to various languages. Starting with Go and Dart. I am trying to parse a simple expressions but the below code always returns NoOp.
```go
package main
import (
	"cuelang.org/go/cue/cuecontext"
	"github.com/davecgh/go-spew/spew"
)
func main() {
	ctx := cuecontext.New()
	const expr = `>30`
	op, vv := ctx.CompileString(expr).Expr()
	// op is always NoOp
	spew.Dump(op, vv)
}
```
Can anyone guide me in the right direction?

**Comment by verdverm** [comment]
It's visible in the `go-spew` output, and would be further down the value tree. I'd expect the value to be a int itself with `>30` as a further constraint, so seeing it nested is not surprising.

There are some other discussions / issues that expand on this same topic, parsing constraints has come up before.

**Reply by srikrsna** [reply]
I tried to use the `Walk` method on the `CompileString` output and it only triggered for itself.

Is there any other way to retrieve the constraints as a struct/interface form similar to an AST?

Edit: I did a quick search with keywords parse and constraint and was not able to find anything relevant.

**Reply by verdverm** [reply]
There is a `Syntax` method on Value that should return the AST. Expect this to get complicated quickly once you start using more CUE language features.

There are some other issues and  discussions on "code gen" that are relevant, not sure on the exact terms to use, finding older issues is a bit difficult now with them being moved over to the new repo. Maybe try the old one too?

**Comment by myitcv** [comment]
@srikrsna this appears to be an edge case (bug?) when compiling the "bare" value `>30`. If you compile CUE that attaches this constraint to a field, then the op is correctly returned.

Please can you raise an issue for your example above?
